src/ol/format/KML.js

```diff
@@ -949,26 +946,29 @@ function createFeatureStyleFunction(style, styleUrl, defaultStyle, sharedStyles,
       if (drawName) {
         name = /** @type {string} */ (feature.get('name'));
         drawName = drawName && !!name;
+        // convert any html character codes
+        if (drawName && name.search(/&[^&]+;/) > -1) {
+          const text = document.createElement('textarea');
```

Comment:
In terms of performance, it would be better to create this `textarea` only once and reuse it.

Reply:
Yes, I've updated it.